yarn-project/circuits.js/src/structs/kernel/combined_accumulated_data.ts

```diff
@@ -77,15 +91,18 @@ export class OptionallyRevealedData {
     /**
      * Events emitted by the function call from which this info originates.
      */
-    public emittedEvents: Fr[],
+    public emittedEvents: Tuple<Fr, typeof EMITTED_EVENTS_LENGTH>,
```

Comment:
It is no longer an array and is instead a large tuple? Why this change?

It is nice that we can specify length here. What is the significance of `typeof` here instead of just length?

Reply:
As discussed, this is how we're representing fixed size arrays

Reply:
Numbers can be both types and values. Typeof converts a number to its type (which can also be the number if its a constant, 5 is a type too)